units/tests/api.rs

```diff
@@ -0,0 +1,227 @@
+//! Test the API surface of `units`.
+//!
+//! The point of these tests are to check the API surface as opposed to test the API functionality.
+//!
+//! What this module tests:
+//!
+//! - The location of re-exports for various typical usage styles.
+//! - Regressions in the API surface (things being accidentally moved).
+//! - All public types implement Debug (C-DEBUG).
+//! - For all non-error types:
+//!     - `Debug` representation is never empty (C-DEBUG-NONEMPTY)
+//! - For all error types:
+//!     - Derive standard traits as defined by `rust-bitcoin` policy.
+//!
+//! This file was created by referring to the output of `cargo check-api`.
+//!
+//! ref: <https://rust-lang.github.io/api-guidelines/about.html>
+
+#![allow(dead_code)]
+#![allow(unused_imports)]
+
+// These imports test "typical" usage by user code.
+use bitcoin_units::locktime::{absolute, relative}; // Typical usage is `absolute::Height`.
+use bitcoin_units::{
+    amount, block, fee_rate, locktime, parse, weight, Amount, BlockHeight, BlockInterval, FeeRate,
+    SignedAmount, Weight,
+};
+
+/// A struct that includes all public non-error enums.
+#[derive(Debug)] // All public types implement Debug (C-DEBUG).
+struct Enums {
+    a: amount::Denomination,
+}
+
+impl Enums {
+    fn new() -> Self { Self { a: amount::Denomination::Bitcoin } }
+}
+
+/// A struct that includes all public non-error structs.
+#[derive(Debug)] // All public types implement Debug (C-DEBUG).
+struct Structs {
```

Comment:
I split this up from `Enums` in case there are more tests we would like to do on just structs (eg Constructors are static, inherent methods ([C-CTOR](https://rust-lang.github.io/api-guidelines/predictability.html#c-ctor)))